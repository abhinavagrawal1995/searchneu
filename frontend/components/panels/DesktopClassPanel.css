/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

.resultsTable {
    border: none !important;
    margin-bottom: 0px !important;
}

.resultsTable th {
    border-left: none !important;
    background: none !important;
}

.inlineBlock {
    display: inline-block;
}

.leftPanel {
    text-align: left;
    display: inline-block;
    line-height: 17px;
    width: calc(100% - 200px);
}

.rightPanel {
    text-align: right;
    display: inline-block;
    float: right;
    white-space: normal;
    max-width: 200px;
    min-width: 53px;
}

.showMoreButton {
    cursor: pointer;
    padding: 10px;
    color: #888888;
    font-weight: 600;
    border-top: 1px solid #ddd;
}

.classGlobeLink {
    opacity: .7;
    float: right;
    display: inline-block;
}

.sectionGlobe {
    opacity: .7;
}

.classTitle {
    display: inline-block;
    max-width: calc(100% - 20px);
}

.classGlobeLinkContainer {
    vertical-align: top;
    display: inline-block;
    float: right;
}

.onlineCheck{
    width: 22px;
}

.onlineHeader{
    width: 78px;
}

.checkIconCell {
    text-align: center !important;
}

.onlineDivLineContainer {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
}

.onlineText {
    /*font-weight: 600;*/
    /*color: rgba(81, 84, 84, 0.75);*/
}

.onlineDivLine {
    vertical-align: middle;
    display: inline-block;
    height: 2px;
    background-color: rgba(9, 10, 10, 0.2);
    width: calc(50% - 102px);
    margin: 21px;
}

.onlineLeftLine {
    margin-left: 36px
}

.wideOnlineCell {
    position: relative;
}

