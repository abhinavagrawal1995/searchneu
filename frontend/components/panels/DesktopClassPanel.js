/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';
import PropTypes from 'prop-types';
import CSSModules from 'react-css-modules';
import classNames from 'classnames/bind';
// import identityObjProxy from 'identity-obj-proxy';

import globe from './globe.svg';
import desktopCss from './DesktopClassPanel.css';
import baseCss from './BaseClassPanel.css';
import macros from '../macros';
import BaseClassPanel from './BaseClassPanel';
import DesktopSectionPanel from './DesktopSectionPanel';


// Merge the base css and the css specific to desktop panels
// The identityObjProxy check is so the class names appear in the snap files in testing
// Lets move to sass instead of css eventually.
const css = {};
Object.assign(css, baseCss, desktopCss);
if (Object.keys(css).length === 0) {
  // css = identityObjProxy
}

const cx = classNames.bind(css);


// Class Panel that renders the box with the class title, class description, and class sections
// If mobile, uses MobileSectionPanel to show the sections.
// The code for desktop is inside this file.


// DesktopClassPanel page component
class DesktopClassPanel extends BaseClassPanel {
  componentDidUpdate() {
    macros.debounceTooltipRebuild();
  }

  componentDidMount() {
    macros.debounceTooltipRebuild();
  }

  // Method to decide whether to show the waitlist or not
  // This logic is different than it is on mobile (because of formatting differences)
  // See MobileSectionPanel.js
  shouldShowWaitlist() {
    const aClass = this.props.aClass;

    // If the class does not have a waitlist, don't show the waitlist
    if (!aClass.getHasWaitList()) {
      return false;
    }

    // If all the sections have 0 seats on the waitlist and 0 total seats, don't show the waitlist (because there isn't actually a waitlist).
    let foundSectionWithWaitlistSeats = false;

    for (const section of aClass.sections) {
      if (section.waitRemaining > 0 || section.waitCapacity > 0) {
        foundSectionWithWaitlistSeats = true;
        break;
      }
    }

    if (!foundSectionWithWaitlistSeats) {
      return false;
    }

    // Also show the waitlist if any of the sections have less than 10 seats left.
    // The number 10 is just an arbitrary decision and can be changed in the future.
    // const foundSectionWithLessThanTenSeats = false;

    for (const section of aClass.sections) {
      if (section.seatsRemaining < 10) {
        return true;
      }
    }

    // If there are plenty of seats left, don't show the waitlist
    return false;
  }

  render() {
    const aClass = this.props.aClass;
    // Render the section table if this class has sections
    let sectionTable = null;
    if (aClass.sections && aClass.sections.length > 0) {
      // Add the Exam column headers if there are any sections in this class that has exam listed
      let examColumnHeaders = null;
      if (aClass.sectionsHaveExam()) {
        examColumnHeaders = [
          <th key='1'>Exam time</th>,
          <th key='3'>Exam date</th>,
          <th key='4'>Exam location</th>,
        ];
      }

      // Add the Online sections head if there are any sections that are online
      // const showOnlineColumn = aClass.getHasOnlineSections();

      const showWaitList = this.shouldShowWaitlist();

      sectionTable = (
        <table className={ `ui celled striped table ${css.resultsTable}` }>
          <thead>
            <tr>
              <th>
                <div className={ css.inlineBlock } data-tip='Course Reference Number'>
                    CRN
                </div>
              </th>
              <th> Professors </th>
              <th> Weekdays </th>
              <th> Time </th>

              <th> Location </th>
              {examColumnHeaders}
              <th> Seats </th>

              <th
                className={ cx({
                  displayNone: !showWaitList,
                }) }
              > Waitlist seats
              </th>
              <th> Link </th>
            </tr>
          </thead>
          <tbody>
            {/* The CSS applied to the table stripes every other row, starting with the second one.
              This tr is hidden so the first visible row is a dark stripe instead of the second one. */}
            <tr style={{ display:'none', paddingTop: 0, paddingBottom: '1px' }} />
            {this.state.renderedSections.map((section) => {
              return <DesktopSectionPanel key={ section.crn } showWaitList={ showWaitList } shouldShowExamColumns={ aClass.sectionsHaveExam() } section={ section } />;
            })}
          </tbody>
        </table>
      );
    }

    // Render the Show More.. Button
    const showMoreSections = this.getShowMoreButton();

    // Figure out the credits string
    const creditsString = this.getCreditsString();

    return (
      <div>
        <div className={ `${css.container} ui segment` }>
          <div className={ css.header }>
            <span className={ css.classTitle }>
              {aClass.subject} {aClass.classId}: {aClass.name}
            </span>
            <span className={ css.classGlobeLinkContainer }>
              <a target='_blank' rel='noopener noreferrer' className={ css.classGlobeLink } data-tip={ `View on ${aClass.host}` } href={ aClass.prettyUrl || aClass.url }>
                <img src={ globe } alt='link' />
              </a>
            </span>
          </div>

          <div className={ css.body }>
            {aClass.desc}
            <br />
            <br />
            <div className={ css.leftPanel }>
              Prerequisites: {this.optionalDisplay(macros.prereqTypes.PREREQ)} {this.showMore(macros.prereqTypes.PREREQ)}
              <br />
              Corequisites: {this.optionalDisplay(macros.prereqTypes.COREQ)} {this.showMore(macros.prereqTypes.COREQ)}
              <br />
              Prerequisite for: {this.optionalDisplay(macros.prereqTypes.PREREQ_FOR)} {this.showMore(macros.prereqTypes.PREREQ_FOR)}
              <br />
              Optional Prerequisite for: {this.optionalDisplay(macros.prereqTypes.OPT_PREREQ_FOR)} {this.showMore(macros.prereqTypes.OPT_PREREQ_FOR)}
            </div>
            <div className={ css.rightPanel }>
              <div data-tip='Check neu.edu for possible updates'> Updated {aClass.getLastUpdateString()}</div>
              {creditsString}
            </div>
          </div>
          {sectionTable}
          {showMoreSections}
        </div>
      </div>
    );
  }
}

// Number of sections to show by default. This is different on mobile.
DesktopClassPanel.sectionsShownByDefault = 3;

DesktopClassPanel.propTypes = {
  aClass: PropTypes.object.isRequired,
};


export default CSSModules(DesktopClassPanel, css);
