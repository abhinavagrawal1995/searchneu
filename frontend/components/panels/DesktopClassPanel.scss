/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

 @import "BaseClassPanel";

.resultsTable {
    border: none !important;
    margin-bottom: 0px !important;
}

.resultsTable th {
    border-left: none !important;
    background: none !important;
}

.inlineBlock {
    display: inline-block;
}

.leftPanel {
    text-align: left;
    display: inline-block;
    width: calc(100% - 200px);
}

.rightPanel {
    text-align: right;
    display: inline-block;
    float: right;
    white-space: normal;
    max-width: 200px;
    min-width: 53px;
}

.showMoreButton {
    cursor: pointer;
    padding: 10px;
    color: #888;
    font-weight: 600;
    border-top: 1px solid #ddd;
}

.classGlobeLink {
    opacity: .7;
    float: right;
    display: inline-block;
}


.classTitle {
    display: inline-block;
    max-width: calc(100% - 20px);
}

.classGlobeLinkContainer {
    vertical-align: top;
    display: inline-block;
    float: right;
}

.onlineCheck{
    width: 22px;
}

.onlineHeader{
    width: 78px;
}

.checkIconCell {
    text-align: center !important;
}


.sectionTableFirstRow {
    display: none;
    padding-top: 0;
    padding-bottom: 1px;
}
