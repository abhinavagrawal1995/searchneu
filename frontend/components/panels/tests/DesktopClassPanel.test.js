/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';


import mockData from './mockData';
import DesktopClassPanel from '../DesktopClassPanel';


// it('should render some stuff', function() {

  const myrenderer = new ShallowRenderer();
  myrenderer.render(<DesktopClassPanel aClass={mockData.cs1210} />);
  const result = myrenderer.getRenderOutput();

  // console.log(result)



    // const component = renderer.create(<DesktopClassPanel aClass={mockData.cs1210} />);
    debugger

    // let tree = component.toJSON()
console.log(result)
    expect(result).toMatchSnapshot();


// });