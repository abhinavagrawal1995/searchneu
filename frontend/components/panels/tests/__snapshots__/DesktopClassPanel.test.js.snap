// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render some stuff 1`] = `
<div>
  <div
    className="undefined ui segment"
  >
    <div
      className={undefined}
    >
      <span
        className={undefined}
      >
        CS
         
        1210
        : 
        Computer Science/Information Science Overview 2: Co-op Preparation
      </span>
      <span
        className={undefined}
      >
        <a
          className={undefined}
          data-tip="View on neu.edu"
          href="https://wl11gp.neu.edu/udcprod8/bwckctlg.p_disp_course_detail?cat_term_in=201810&subj_code_in=CS&crse_numb_in=1210"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="link"
            src="file"
          />
        </a>
      </span>
    </div>
    <div
      className={undefined}
    >
      Continues the preparation of students for careers in the computing and information fields by discussing co-op and co-op processes. Offers students an opportunity to prepare a professional résumé; practice proper interviewing techniques; explore current job opportunities; learn how to engage in the job and referral process; and to understand co-op policies, procedures, and expectations. Discusses professional behavior and ethical issues in the workplace. 1.000 Lecture hours
      <br />
      <br />
      <div
        className={undefined}
      >
        Prerequisites: 
        <span
          className="dynamicText"
        >
          None
        </span>
         
        <br />
        Corequisites: 
        <span
          className="dynamicText"
        >
          None
        </span>
         
        <br />
        Prerequisite for: 
        <span
          className="dynamicText"
        >
          None
        </span>
         
        <br />
        Optional Prerequisite for: 
        <span
          className="dynamicText"
        >
          None
        </span>
         
      </div>
      <div
        className={undefined}
      >
        <div
          data-tip="Check neu.edu for possible updates"
        >
           Updated 
          6 days ago
        </div>
        1 credits
      </div>
    </div>
    <table
      className="ui celled striped table undefined"
    >
      <thead>
        <tr>
          <th>
            <div
              className={undefined}
              data-tip="Course Reference Number"
            >
              CRN
            </div>
          </th>
          <th>
             Professors 
          </th>
          <th>
             Weekdays 
          </th>
          <th>
             Time 
          </th>
          <th>
             Location 
          </th>
          <th>
             Seats 
          </th>
          <th
            className="displayNone"
          >
             Waitlist seats
          </th>
          <th>
             Link 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          style={
            Object {
              "display": "none",
              "paddingBottom": "1px",
              "paddingTop": 0,
            }
          }
        />
        <tr>
          <td>
             
            13502
             
          </td>
          <td>
             
            Jennifer Anne Shire
             
          </td>
          <td>
             
            <div
              className="inlineBlock daysContainer"
              data-tip="Meets on Tuesday"
            >
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBoxChecked weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
            </div>
             
          </td>
          <td>
            9:50 - 11:30 am
          </td>
          <td>
             
            <span>
              <span>
                <a
                  href="https://maps.google.com/?q=Northeastern University Snell Engineering Center "
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Snell Engineering Center 108
                </a>
              </span>
            </span>
             
          </td>
          <td>
            <div
              className={undefined}
              data-tip="Open Seats/Total Seats"
            >
              0
              /
              19
            </div>
          </td>
          <td
            className="displayNone"
          >
            <div
              className={undefined}
              data-tip="Open/Total Waitlist Seats"
            >
              0
              /
              0
            </div>
          </td>
          <td>
            <a
              className="undefined undefined"
              data-tip="View on neu.edu"
              href="https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=201810&crn_in=13502"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="link"
                src="file"
              />
            </a>
          </td>
        </tr>
        <tr>
          <td>
             
            14404
             
          </td>
          <td>
             
            Jennifer Anne Shire
             
          </td>
          <td>
             
            <div
              className="inlineBlock daysContainer"
              data-tip="Meets on Tuesday"
            >
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBoxChecked weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
            </div>
             
          </td>
          <td>
            11:45 - 1:25 pm
          </td>
          <td>
             
            <span>
              <span>
                <a
                  href="https://maps.google.com/?q=Northeastern University Ryder Hall "
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Ryder Hall 431
                </a>
              </span>
            </span>
             
          </td>
          <td>
            <div
              className={undefined}
              data-tip="Open Seats/Total Seats"
            >
              0
              /
              19
            </div>
          </td>
          <td
            className="displayNone"
          >
            <div
              className={undefined}
              data-tip="Open/Total Waitlist Seats"
            >
              0
              /
              0
            </div>
          </td>
          <td>
            <a
              className="undefined undefined"
              data-tip="View on neu.edu"
              href="https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=201810&crn_in=14404"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="link"
                src="file"
              />
            </a>
          </td>
        </tr>
        <tr>
          <td>
             
            14405
             
          </td>
          <td>
             
            Jennifer Anne Shire
             
          </td>
          <td>
             
            <div
              className="inlineBlock daysContainer"
              data-tip="Meets on Tuesday"
            >
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBoxChecked weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
              <div
                className="weekDayBox"
              />
            </div>
             
          </td>
          <td>
            11:45 - 1:25 pm
          </td>
          <td>
             
            <span>
              <span>
                <a
                  href="https://maps.google.com/?q=Northeastern University Ryder Hall "
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Ryder Hall 431
                </a>
              </span>
            </span>
             
          </td>
          <td>
            <div
              className={undefined}
              data-tip="Open Seats/Total Seats"
            >
              0
              /
              19
            </div>
          </td>
          <td
            className="displayNone"
          >
            <div
              className={undefined}
              data-tip="Open/Total Waitlist Seats"
            >
              0
              /
              0
            </div>
          </td>
          <td>
            <a
              className="undefined undefined"
              data-tip="View on neu.edu"
              href="https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=201810&crn_in=14404"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="link"
                src="file"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="showMoreButton"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      Show More...
    </div>
  </div>
</div>
`;
